scio-jdbc/src/main/scala/com/spotify/scio/jdbc/JdbcOptions.scala

```diff
@@ -71,19 +72,28 @@ sealed trait JdbcIoOptions
  *   use apache beam default fetch size if the value is -1
  * @param outputParallelization
  *   reshuffle result to distribute it to all workers. Default to true.
+ * @param dataSourceProviderFn
+ *   function to provide a custom [[javax.sql.DataSource]]
  */
 final case class JdbcReadOptions[T](
   connectionOptions: JdbcConnectionOptions,
   query: String,
   statementPreparator: PreparedStatement => Unit = null,
   rowMapper: ResultSet => T,
   fetchSize: Int = JdbcIoOptions.BeamDefaultFetchSize,
-  outputParallelization: Boolean = JdbcIoOptions.DefaultOutputParallelization
+  outputParallelization: Boolean = JdbcIoOptions.DefaultOutputParallelization,
+  dataSourceProviderFn: () => DataSource = null,
+  configOverride: Read[T] => Read[T] = identity
```

Comment:
Do we agree to apply this convention on all IOs for 0.13 ?

Reply:
@RustedBones it would be good! We need to be consistent across all APIs